### PR TITLE
[DOCS-7630] Add Keycloak Tutorial to ACS 7.3+

### PR DESCRIPTION
**Pull request by mbujak-hyland**

### CODE REVIEW

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:244]:
```suggestion
| identity-service.auth-server-url |Keycloak's base URL, for example `https://keycloak.example.com/auth`|
| identity-service.realm |The realm name configured in Keycloak for the Alfresco application, for example `alfresco`|
| identity-service.resource |The **Client ID** set up in the Desktop Sync for Alfresco Content Services. The client needs to exist underneath the realm set for `identity-service.realm`, for example `desktop-sync`|
| identity-service.public-client |The adapter will not send credentials for the client to Keycloak if this is set to true, for example `true`|
| identity-service.credentials.secret |The secret key for this client if the access type is not set to public.|
```

Again, these property keys can't be updated to something that doesn't reflect the current status of the software. They have to be reverted to their original variant.

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:248]:
```suggestion
Alfresco Process Services (APS) has two sets of properties that need to be configured to setup SSO. One set synchronizes APS with an LDAP directory and the other set configures the connection with Keycloak.
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:265]:
```suggestion
2. Configuration with the Alfresco Process Services can be achieved manually for WAR file deployments using the `activiti-identity-service.properties` or reference an external file for Docker and Kubernetes deployments:
```

Same reasoning as per other similar changes, cc: @balsarori in case something changed here and I'm not aware.

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:269]:
```suggestion
    | keycloak.enabled |Sets whether Process Services will use Keycloak to authenticate against, for example `true`|
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:270]:
```suggestion
    | keycloak.realm |The realm name configured in Keycloak for the Alfresco applications, for example `alfresco`|
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:271]:
```suggestion
    | keycloak.auth-server-url |Keycloak's base URL, for example `https://keycloak.example.com/auth`|
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:273]:
```suggestion
    | keycloak.resource |The **Client ID** set up in Keycloak for Process Services. The client needs to exist underneath the realm set for `keycloak.realm` or `KEYCLOAK_REALM`, for example `alfresco`|
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:275]:
```suggestion
    | keycloak.public-client |The adapter will not send credentials for the client to Keycloak if this is set to `true`, for example `true`|
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:279]:
```suggestion
    | keycloak.enable-basic-auth |Sets whether basic authentication is also supported by Keycloak, for example `true`|
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:285]:
```suggestion
1. Set these additional properties in `activiti-identity-service.properties`:
```
Same as above, cc: @balsarori.

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:315]:
```suggestion
If Process Services for mobile is required then a client needs to be created for it in Keycloak to enable SSO capability. The redirect URI is preconfigured for the mobile application using the operating system it is installed on, which means that the **Valid Redirect URIs** value in Keycloak must match this value.
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:334]:
```suggestion
If Content Services for iOS is required then a client needs to be created for it in Keycloak to enable SSO capability. The redirect URI is preconfigured for the mobile application using the operating system it is installed on, which means that the **Valid Redirect URIs** value in Keycloak must match this value.
```

File: content-services/latest/tutorial/sso/saml.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/saml.md:351]:
```suggestion
> **Note:** If timeout is configured using the same value for Keycloak and ACS, accessing any of the applications after the specified time will prompt a user to sign in again.
```

**Review by dsibilio — CHANGES_REQUESTED** [review]
Reviewed all files except `kerberos.md` and `ldap.md`. I believe there are quite a few things to change so I'll stop here until they're addressed and my review is re-requested.

Thanks for taking care of this 🙏

File: _data/toc/content-services-tutorial.yaml
Comment (mbujak-hyland) [reply _data/toc/content-services-tutorial.yaml:21]: It is just for now, once you approve the changes in ACS 23.x, I will just copy the content to ACS 7.3 and 7.4.

File: content-services/latest/tutorial/sso/ldap.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/ldap.md:211]: The original documentation stated:

`The client needs to exist underneath the realm set for keycloak.realm or IDENTITY_SERVICE_REALM, for example alfresco`

@balsarori @jamalkm was this a mistake in the first place? Was this supposed to be _"`keycloak.realm` or `KEYCLOAK_REALM`"_ in the first place?

I assume the new documentation wording makes sense if we're referring to [binding from environment variables](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.relaxed-binding.environment-variables) via Spring, I assume it wouldn't if there is a custom mechanism to specifically look for an `IDENTITY_SERVICE_REALM` env var.

@mbujak-hyland please wait for @balsarori or @jamalkm confirmation related to this question.

File: content-services/latest/tutorial/sso/ldap.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/ldap.md:252]: @balsarori does this change make sense?

File: content-services/latest/tutorial/sso/ldap.md
Comment (dsibilio) [inline content-services/latest/tutorial/sso/ldap.md:296]:
```suggestion
> **Note:** If timeout is configured in [Keycloak](#step-1-configure-a-realm-and-clients), accessing any of the applications after the specified time will prompt a user to sign in again.
```

Comment (dsibilio) [reply content-services/latest/tutorial/sso/ldap.md:296]: Typo and punctuation

**Review by dsibilio — CHANGES_REQUESTED** [review]
Left a few more considerations and added @balsarori as a reviewer to answer to some of the questions & double-check that the APS properties are up-to-date and still make sense after the PR.

File: content-services/latest/tutorial/sso/ldap.md
Comment (jamalkm) [reply content-services/latest/tutorial/sso/ldap.md:211]: there used to be a script in APS that  checks if the environment variable IDENTITY_SERVICE_REALM is set to a non-empty value. If it is, it replaces the keycloak.realm value with that one.

@dsibilio there used to be a script in APS that checks whether the environment variable `IDENTITY_SERVICE_REALM` is set to a non-empty value. If it is, it replaces the `keycloak.realm` value with that one. This is for @balsarori to confirm.

Comment (dsibilio) [reply content-services/latest/tutorial/sso/ldap.md:211]: @moberoi2

File: content-services/latest/tutorial/sso/ldap.md
Comment (dsibilio) [reply content-services/latest/tutorial/sso/ldap.md:252]: Yes. But is it "Keycloak authentication" within the APS UI? Or is it "Identity Service authentication" as mentioned by the already existing SSO guide: https://docs.alfresco.com/identity-service/latest/tutorial/sso/ldap/#step-8-optional-configure-a-connection-between-process-services-and-content-services?

File: content-services/latest/tutorial/sso/ldap.md
Comment (dsibilio) [reply content-services/latest/tutorial/sso/ldap.md:212]: **Is there a DOCS ticket for this change? Why hasn't it been reflected into the official SSO guide yet?** 

**Please coordinate with and guide @mbujak-hyland through the necessary changes** that this will require both for the "latest" (23.x) version of ACS and for previous ones depending on their supported platform's stated supported version of APS. E.g.: if ACS 7.3 and ACS 7.4 do not list APS 24.x as supported platform then the properties as they were in APS 2.x should be part of the docs there. Whereas for the latest we should have the new APS docs.

File: content-services/latest/tutorial/sso/ldap.md
Comment (moberoi2) [reply content-services/latest/tutorial/sso/ldap.md:252]: Keycloak Authentication is done via Identity service management . It is not within APS UI.

Comment (dsibilio) [reply content-services/latest/tutorial/sso/ldap.md:252]: The instructions state:

```
2. Sign into Process Services as an administrator.

3. Navigate to **Identity Management** > **Tenants** > **Alfresco Repositories**.
```

then further down within the same paragraph, they state:

```
    | Setting | Description |
    | ------- | ----------- |
    | Name | A name for the repository connection. |
    | Alfresco tenant | The tenant to create the repository under. |
    | Repository base URL | The base URL of the repository instance to connect to. |
    | Share base URL | The base URL of Share for the repository instance to connect to. |
   [...]
```

This can't refer to Keycloak's UI either as it contains Alfresco-specific notions, so what is this _"Identity service management"_ UI you're talking about? 🤔 

**If we're unsure, then I'd suggest retaining the "Identity Service authentication" wording for this line instead of switching to "Keycloak authentication".**

File: content-services/latest/tutorial/sso/ldap.md
Comment (balsarori) [reply content-services/latest/tutorial/sso/ldap.md:211]: There is no `KEYCLOAK_REALM` env var. In APS 2, there was a property called `keycloak.realm` which is set by setting the env var `IDENTITY_SERVICE_REALM`. In APS 24.x the property name was changed to `activiti.identity-service.realm` the env var remains `IDENTITY_SERVICE_REALM`

Comment (dsibilio) [reply content-services/latest/tutorial/sso/ldap.md:211]: Quoting @balsarori:

> there is no `KEYCLOAK_REALM` env var. In APS 2, there was a property `keycloak.realm` which is set by setting the `IDENTITY_SERVICE_REALM`. In APS the property name was changed to `activiti.identity-service.realm`

@mbujak-hyland I also confirmed that both for APS 2 and APS 24 the env var is always `IDENTITY_SERVICE_REALM` and can't be renamed in the docs to `KEYCLOAK_REALM`

File: content-services/latest/tutorial/sso/ldap.md
Comment (moberoi2) [reply content-services/latest/tutorial/sso/ldap.md:252]: For Identity service management, I meant Identity Service Authentication only. Yes retaining Identity Service Authentication is feasible

Comment (dsibilio) [reply content-services/latest/tutorial/sso/ldap.md:252]: With @balsarori I have confirmed that this is actually referring to an APS UI and the wording never changed, so this paragraph needs to state **"Identity Service authentication**" instead of **"Keycloak authentication"** @mbujak-hyland

File: content-services/7.4/tutorial/sso/ldap.md
Comment (balsarori) [inline content-services/7.4/tutorial/sso/ldap.md:207]: For which version of APS is this document for? These properties will work for APS 2.x and will not for APS 24.x

File: content-services/7.4/tutorial/sso/saml.md
Comment (balsarori) [inline content-services/7.4/tutorial/sso/saml.md:269]: Same here, what version of APS?

File: content-services/7.4/tutorial/sso/ldap.md
Comment (mbujak-hyland) [reply content-services/7.4/tutorial/sso/ldap.md:207]: Hi @balsarori, this is for ACS 7.4 which is compatible with APS 2.x. For ACS 23.x (which is compatible with APS 24.x) these properties are different: mainly "keycloak" is replaced with "activiti.identity-service". Is it correct?

**Review by balsarori — APPROVED** [review]
LGTM

**Review by mbujak-hyland — COMMENTED** [review]
OK

File: _data/toc/content-services-tutorial.yaml
Comment (anxumalo) [reply _data/toc/content-services-tutorial.yaml:21]: @mbujak-hyland Copy the 'Configure Keycloak' to the ACS 7.3 & 7.4 sections as well so they appear in the TOC

File: content-services/7.4/tutorial/sso/ldap.md
Comment (anxumalo) [inline content-services/7.4/tutorial/sso/ldap.md:105]: Similar to the previous comment, make sure all links within `content-services/7.4/` remain in the same folder and not `content-services/latest/`. Check for other instances under `content-services/7.4/`.

**Review by anxumalo — CHANGES_REQUESTED** [review]

**Review by anxumalo — APPROVED** [review]
LGTM

**Pull request merged**
